### ISSUE TITLE
How to use snapshots with btrfs?

### ISSUE THREAD

**Issue by pierrechevalier83** [picodotdev/alis#140]
Hi, I've discovered alis a few days ago and I'm impressed by the quality of the script and the sane defaults all around.

There is one thing I haven't managed to get to work with the script as-is, though; and I'm wondering whether it's because I'm missing something or if the script has a bug.

As is, the script creates btrfs subvolumes: `/root`, `/var`, `/home` and `/snapshots` mounted at `/`, `/var`, `/home` and `/snapshots` respectively.

For me, this setup does not work with [snapper](https://wiki.archlinux.org/title/Snapper) for snapshots as that tool relies on the snapshots being at `/.snapshots` on the filesystem rather than `snapshots`.

Am I missing a config option of snapper that would allow me to point it at `/snapshots` instead of `/.snapshots`? Or should `alis.sh` mount the `/snapshots` subvolume at the `/.snapshots` directory instead?

**Comment by picodotdev** [comment]
Hello Pierre, I am happy you found useful this script :).

I am not btrfs and Snapper expert but reading the wiki [1] about it you should execute this command. The .snapshots subvolume inside snapshoted subvolume is created by snapper create-config command.

$ sudo snapper -c snapshots create-config /snapshots

This should create a /snapshots/.snapshots subvolume. But I think you maybe want to create snapshots of other subvolumes like boot, home, var. For home should be, this will create /home/.snapshots subvolume.

$ sudo snapper -c home create-config /home

Let me know if this solves your question.

[1]: https://wiki.archlinux.org/title/Snapper

**Comment by pierrechevalier83** [comment]
Hi @picodotdev, thank you for your prompt reply.

I can see I didn't give enough context in the OP. What I was trying to achieve is backing up the root partition so I can restore a prior version of the system with `snapper-rollback`, in case I mess it up.
 I was reading the Snapper wiki page, and in particular, the "Suggested filesystem layout" section. It seems to recommend having the `/.snapshots` directory to live on its own subvolume which is a child of subvolid=5.
 
 When I saw a `snapshots` subvolume in `alis.sh`, I had a bit of tunnel vision and assumed it was meant for this specific context.
 
 I ended up managing to do what I wanted to do without relying on modifications in the alis script, by performing [a couple of tweaks](https://github.com/pierrechevalier83/alis-conf/blob/39d951ede7b73fc8cc7eef7c27164a39c2eac874/post_install.sh#L84) after installation, including pointing the fstab entry for snapshots at the `/.snapshots` directory.
 
 Since it seems I misunderstood the intent of the `snapshots` subvolume in `alis.sh`, feel free to close this ticket.
 
 Thanks again for the clarification :smile:
